### PR TITLE
General bugfixes - Gandalf2k15

### PR DESCRIPTION
**Pull request by Gandalf2k15**
# **Bugfixes and Features**

Okay, here is a quick list of things fixed:
- Gateway: Maploader
- Outpost: Fixes shuttle, caused runtimes.
- Outpost: Adds a new section, workshop.
- Outpost: Edits power systems, sixty solar panels instead of 15.
- Removes a ripley.
- Removes autolathe

Gateway: 
Fairly simple update/bugfix, taken from paradise. Also disables the gateway until further notice. Sadly the bugfixes to maps are considered 'balancing'. So I'm not including it here.

Outpost:
New solar section:
Image of Solars
New workshop section:
Image of  workshop

All systems have been tested three times.

### CODE REVIEW

**Comment by Carbonhell** [comment]
i'm completely against the idea of R&D on the outpost,which will probably cause scientists to get all the endgame research loot and then just hop back to the station and use it to disrupt the round's flow
that's my thought

**Comment by trifornt-tinfornt** [comment]
Well, what's stopping them from just going to Tech Storage and getting the parts, or deconning the normal ones in RnD?

**Comment by Gandalf2k15** [comment]
The parts are gonna be in engi storage, for them. I'll update pics soon

**Comment by trifornt-tinfornt** [comment]
Not a fan of the roundstart Ripley honestly. It'd just make Malf AI even harder to manage, as Ripleys destroy the AI without a second thought - and these ones can't be prevented by just killing the roboticists.

**Comment by Gandalf2k15** [comment]
Its just the same as the miners one.

**Comment by ghost** [comment]
Tbh, science can easily make an ripley from roundstart for mining anyway. I don't see the need for them to start with one.

**Comment by Gandalf2k15** [comment]
Updated image.

**Comment by calibraptor** [comment]
The science outpost doesn't need R&D nor a ripley.

Are you trying to completely invalidate miners?

:-1:

**Comment by Gandalf2k15** [comment]
The ripley is in place because most scientists kept on blowing the station to shit using bombs, this gives them an alternate. This changes almost nothing because science can simply make one anyway. Just stops them from blowing up the station because they are lazy.

**Comment by Gandalf2k15** [comment]
CALIB READ WHY ITS THERE

**Comment by calibraptor** [comment]
My reasoning still stands. The science outpost gets mining gear and bombs to mine with, all a scientist has to do is walk a short distance away from the outpost before setting off a bomb.

**Comment by Gandalf2k15** [comment]
Updated reasoning.

**Comment by calibraptor** [comment]
Increasing the time it takes to get R&D up and running by about 30 seconds doesn't change that fact that you're trying to give the outpost R&D. The outpost doesn't need R&D, it already acts as a backup xenobiology and toxins lab.

**Comment by Gandalf2k15** [comment]
It does though. Read the reasoning, I might actually start playing scientist again

**Comment by Gandalf2k15** [comment]
I just tested it. It takes around 1 minute and 30 seconds to set up. Which is actually a long time if you look at the length of some hippie rounds.

**Comment by calibraptor** [comment]
If R&D is added to the outpost scientists will just hole up in it like it's their own personal autism fort every single round, there will be no reason to do R&D on the station because on the outpost you don't have to worry about people breaking in or antagonists.

Firm -1

Remove the ripley and remove the R&D.

**Comment by ghost** [comment]
I'd rather R&D stay on-station and that the ripley not exist on the outpost at all. -1 to those, +1 to the other changes.

**Comment by Gandalf2k15** [comment]
I still strongly disagree with you. The reasoning is super simple. Science is utilised. Everyone is happy. Stop thinking about antagonists. Its barely an autism fort, its the same level of autism as the mining station.

**Comment by Gandalf2k15** [comment]
Also, R&D does not get removed from on station. This is an Alternate that takes effort to set up.

**Comment by ghost** [comment]
Yeah, I'm aware that it doesn't remove R&D from the station physically, but it might as well do so. Though it requires setup, scientists would probably rather fuck off to a more secluded and secure area than stay on the station and have to share dangers with the rest of the crew. It's simply more logical for them to move out than stay in.

**Comment by calibraptor** [comment]
Literally what Dudeinyourbackyard just said. Also the science outpost is very much an autism fort considering it's all r-walls and there's shutters on every window.

**Comment by Gandalf2k15** [comment]
But you have not given a solid reason to -1 it, all you are really outlining is what would happen. None of this causes balancing issues or anything. Its just more convenient for the scientists.

**Comment by calibraptor** [comment]
The balancing issue is that scientists wouldn't have to interact with the rest of the crew whatsoever if this is merged as is with R&D and a ripley.

**Comment by Nebyab** [comment]
I can't fathom how this is a good idea. What's the advantage of this beyond further isolating scientists from the round? What is this supposed to add to the game? 

-1

**Comment by Gandalf2k15** [comment]
Nebyab, again. Soo vague. Whats the problem with having isolated people for like 5 minutes. Seriously. Think about mining, they are away for the entire round. God sake, its like I'm trying to delete the world or something. This will mean R&D ACTUALLY gets done, it would give people a reason to fucking do science. It would stop people from fucking off when they roll scientist... Jesus christ I don't understand why you are soo against having R&D on the outpost.... It only makes the game better.

**Comment by doobiedoo23** [comment]
why should you give a whole extra research area to scientists? one is enough, all this'll do is give antags and powergamers a safe place to max research and go fuck up the station

**Comment by Gandalf2k15** [comment]
Why not. Why not let science actually get done, why not let scientists have fun why not?

**Comment by calibraptor** [comment]
The science outpost was added to allow scientists to mine their own ore if mining isn't doing their fucking job. R&D is easier now than it's ever been before, I have no idea why you think it needs to be even easier.

**Comment by Gandalf2k15** [comment]
I'll just sum up the problems with R&D.
PROS:
- This will enable people to complete science to its full potential.
- This will give scientists the intensive to do their job.
- This gives the outpost more functionality
- This truly makes science independent of mining.

CONS:
- This would likely cause several people to isolate themselves from the main station, however I see this as a non issue due to the fact its their own choice. However this could potentially fuck over antagonists.
- This would make science to over powered as such, I have attempted to combat this by making all of the machines inoperable without intervention.

Balancing is something that I'm really not good at, however I have at least made an attempt. I would ask that @Nebyab please add to this list, I really want to make this work out. I can't do that when people are against me, infact I would like everyone to look over this list and amend it.

**Comment by dreamcrab** [comment]
no

**Comment by dreamcrab** [comment]
no no no no no

**Comment by dreamcrab** [comment]
another no for good measure

**Comment by Gandalf2k15** [comment]
aaaaaaaaaaaaaaaaaa reasoning pls

**Comment by doobiedoo23** [comment]
no

**Comment by dreamcrab** [comment]
because i didnt want to take even a little bit of science off the station in the first place and now youre completely moving it off station

no no no no no no no no no

**Comment by ghost** [comment]
Science has ez access to the asteriod and can easily make ripleys with a few minutes of work.
RnD should be for the good of the station and this encourages science to not share. (There was a reason the ore redemption on the mining post was decommissioned and the one on station was re-added.)
Was the satellite running out of power or some shit? Is the extra set up meme solar panels required?

**Comment by Gandalf2k15** [comment]
Okay, well I understand. I'll remove R&D and leave a blank room. And Mcbaw, it was slow. You do or don't want the ripley there?

**Comment by dreamcrab** [comment]
NO RIPLEY OBVIOUSLY LIKE WE SAID LAST TIME YOU TRIED TO PUT IT IN

**Comment by dreamcrab** [comment]
DONT HAVE EMPTY ROOMS EITHER

**Comment by ghost** [comment]
If you want a ripley as science, you can easily research it, break into robotics and make it.

**Comment by Gandalf2k15** [comment]
I'll relable this room to equipment room.

**Comment by Nebyab** [comment]
Let's go down this rank by rank.

Pros
1. Science being done every round _isn't_ a good thing. Other servers _literally have problem with_ people maxing R&D every single round.
2. How exactly does being able to do R&D in deep space incentivize people to do their job, more than doing it on the station. According to everyone bar you, R&D gets done nearly every round.
3. That's a fact, not a feature. "More shit" Isn't always a pro.
4. This is not a good thing either. "LESS INTERDEPARTMENT INTERACTION HIP-HIP-HOORAY"

Cons
1. This has nothing to do with antagonists. This is a multiplayer game. Miners are alone, simply because there is no other way to do it, not because we're trying to compartmentalize everyone into utter isolation in the name of...? I... I don't know. R&D being maxed more often I guess?

and I needn't go over the others

**Comment by Gandalf2k15** [comment]
Granted, I don't see it as an issue. However thats my personal opinion. I will remove R&D and the ripley, replaced it with an actually cool looking workshop.

**Comment by Gandalf2k15** [comment]
Okay, this has been updated. Removes the R&D section and ripley! Wew. Please amend votes accordingly.

**Comment by calibraptor** [comment]
Update screenshots.

**Comment by Gandalf2k15** [comment]
wheps

**Comment by calibraptor** [comment]
> Autolathe

Fuck man seriously?

**Comment by Gandalf2k15** [comment]
Its called a workshop for a reason. Let me fix my completely gamembreaking bugs first.

**Comment by calibraptor** [comment]
Remove the autolathe and don't add anymore stupid shit and you'll have a +1 from me.

**Comment by Gandalf2k15** [comment]
alrighty

**Comment by calibraptor** [comment]
Why not just remove the useless extra room btw?

**Comment by Gandalf2k15** [comment]
Because solars.

**Comment by Gandalf2k15** [comment]
Also updated.

File: code/modules/awaymissions/gateway.dm [hunk 1]
```diff
@@ -95,9 +96,13 @@ var/obj/machinery/gateway/centerstation/the_gateway = null
 
 
 /obj/machinery/gateway/centerstation/proc/toggleon(mob/user)
-	if(!ready)			return
-	if(linked.len != 8)	return
-	if(!powered())		return
+	user << "<span class='notice'>Good luck! Please give us feedback on the map you played: http://hippiestation.com/showthread.php?tid=4012</span>"
```
Comment (Carbonhell) [inline code/modules/awaymissions/gateway.dm:99]: Should go after the safety checks.

File: code/modules/awaymissions/gateway.dm [hunk 2]
```diff
@@ -200,8 +213,11 @@ var/obj/machinery/gateway/centerstation/the_gateway = null
 
 
 /obj/machinery/gateway/centeraway/proc/toggleon(mob/user)
-	if(!ready)			return
-	if(linked.len != 8)	return
+	user << "<span class='notice'>Well done! Please give us feedback on the map you played: http://hippiestation.com/showthread.php?tid=4012</span>"
```
Comment (Carbonhell) [inline code/modules/awaymissions/gateway.dm:216]: Same here,should go after the safety checks

File: code/modules/awaymissions/maploader/reader.dm [hunk 3]
```diff
@@ -81,23 +78,25 @@ var/global/dmm_suite/preloader/_preloader = null
 
 			//fill the current square using the model map
 			xcrd=0
-			for(var/mpos=1;mpos<=x_depth;mpos+=key_len)
+
+			for(var/mpos in 1 to x_depth step key_len)
```
Comment (Carbonhell) [inline code/modules/awaymissions/maploader/reader.dm:82]: huh what's step?

**Comment by Carbonhell** [comment]
Code will be ok after you fix what i pointed out,but next time please keep bugfixes in a separate PR than features, thank you.

**Comment by Gandalf2k15** [comment]
aighty

**Comment by calibraptor** [comment]
The science outpost has been removed by popular vote. Hence this PR now conflicts.

**Pull request closed without merging**